From 985b3d570cc0ee50d33c6306397b57de7941cd80 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Mon, 21 Dec 2020 17:30:02 +0100
Subject: [PATCH 0699/1010] evl/thread: add evl_current_kthread()

Returns the current kthread descriptor or NULL if another thread
context is running. CAUTION: does not account for IRQ context.

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/evl/thread.h | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/include/evl/thread.h b/include/evl/thread.h
index 4cdc0479c5..aa34855f05 100644
--- a/include/evl/thread.h
+++ b/include/evl/thread.h
@@ -397,4 +397,13 @@ int evl_join_kthread(struct evl_kthread *kthread,
 	return evl_join_thread(&kthread->thread, uninterruptible);
 }
 
+static inline struct evl_kthread *
+evl_current_kthread(void)
+{
+	struct evl_thread *t = evl_current();
+
+	return !t || t->state & T_USER ? NULL :
+		container_of(t, struct evl_kthread, thread);
+}
+
 #endif /* !_EVL_THREAD_H */
-- 
2.30.2

